server/datastore/mysql/hosts.go

```diff
@@ -489,6 +490,17 @@ const hostMDMSelect = `,
 	END as mdm_server_url
 	`
 
+func postFillHost(host *fleet.Host) {
+	host.MDM.EnrollmentStatus = host.DBOnlyMDMEnrollmentStatus
+	host.MDM.ServerURL = host.DBOnlyMDMServerURL
+}
+
+func postFillHosts(hosts []*fleet.Host) {
```

Comment:
I think this makes sense to do as post-processing to the DB query (especially since the hosts query is already so complex) but I was also wondering about the possibility of using `JSON_OBJECT` instead as a way to cut out the extra loop. Of course, that would mean we would need to loop instead when we flatten the results for the csv endpoint, but that endpoint presumably is called less frequently (and I imagine a bit of delay in the UX on the report download would seem like less of a drag than in the GUI). This is probably premature optimization on my part so I don't think we need to change this PR, but I just wanted to raise the topic to hear your perspective :)

Reply:
You mean to return a JSON object with the mdm enrollment status and server url into a single host struct field? That's interesting, I didn't know the DB driver supported that - if it works as I think it does, we could do without the post-processing and I _think_ the CSV would still work as-is (i.e. if the query's JSON_OBJECT field can unmarshal directly into the `MDMHostData` struct, then it would work the same as it does for this PR, even the CSV would work).

I'll give it a shot see if that could work.

That being said, the loop here is not a performance concern - this is a loop on objects that are already in memory, it would basically be unnoticeable vs the query and network times even if we queried 100s or 1000s of hosts (we typically query 20 per page, I believe).

Reply:
@gillespi314 Thanks for the suggestion, it turned out much cleaner than having the extra db-only fields and post-processing! Decoding JSON from the DB was not supported out-of-the-box, but it was a simple custom [sql.Scanner](https://pkg.go.dev/database/sql#Scanner) implementation away.